server/src/interfaces/idb.py

```diff
@@ -0,0 +1,79 @@
+"""
+We use formal interfaces to enforce **modularity** first and foremost, and then structure
+onto all of the code that is to be written.
+
+Think of this as mathematical axioms, with which you then build proofs which you then use
+together to build new proofs.
+"""
+import abc
+from typing import Any, Dict, List, Optional
+
+# todo: some of the items below can raise. Write docs for it
+
+
```

Comment:
let's also add a todo here and in the other file, that it is being repeated and we should do something about it